uaclient/contract.py

```diff
@@ -149,14 +149,17 @@ def request_machine_token_refresh(
 
         @return: Dict of the JSON response containing refreshed machine-token
         """
+        os = util.get_platform_info()
+        arch = os.pop("arch")
         if not machine_id:
             machine_id = util.get_machine_id(self.cfg.data_dir)
         headers = self.headers()
         headers.update({"Authorization": "Bearer {}".format(machine_token)})
         url = API_V1_TMPL_CONTEXT_MACHINE_TOKEN_REFRESH.format(
             contract=contract_id, machine=machine_id
         )
-        response, headers = self.request_url(url, headers=headers)
+        data = {"machineId": machine_id, "architecture": arch, "os": os}
```

Comment:
Our since our contract client provides content-type': 'application/json', serviceclient.py  json.dumps the dict structure and encodes it as a utf-8 bytestring which it sends to the ua-contracts backend:
 URL [POST]: https://contracts.staging.canonical.com/v1/context/machines/token, headers: {'content-type': 'application/json', 'accept': 'application/json', 'user-agent': 'UA-Client/19.6~646~git.96620f1~ubuntu14.04.1', 'Authorization': 'Bearer <REDACTED>'}, data: b'{"machineId": "<REDACTED>", "architecture": "x86_64", "os": {"distribution": "Ubuntu", "series": "trusty", "kernel": "4.4.0-148-generic", "type": "Linux", "version": "14.04 LTS (Trusty Tahr)", "release": "14.04"}}'